groupease-server/src/main/java/io/github/groupease/group/GroupWebService.java

```diff
@@ -0,0 +1,93 @@
+package io.github.groupease.group;
+
+import javax.annotation.Nonnull;
+import javax.ws.rs.*;
+import javax.ws.rs.core.MediaType;
+
+import com.codahale.metrics.annotation.Timed;
+import javax.inject.Inject;
+
+import io.github.groupease.model.Group;
+import io.github.groupease.model.Member;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.lang.invoke.MethodHandles;
+import java.util.List;
+
+/**
+ * REST-ful web service for {@link Group} instances.
+ */
+@Path("groups")
+@Produces(MediaType.APPLICATION_JSON)
+public class GroupWebService {
+    private static final Logger LOGGER = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+
+    private final GroupDao groupDao;
+
+    @Inject
+    public GroupWebService(@Nonnull GroupDao groupDao)
+    {
+        this.groupDao = groupDao;
+    }
+
+    public List<Group> listInChannel(long channelId)
+    {
+        LOGGER.debug("GroupWebService.listInChannel({})", channelId);
+        //Todo: auth check that caller is member of channel
+
+        return groupDao.listInChannel(channelId);
+    }
+
+    @GET
+    @Path("{id}")
+    @Timed
+    @Nonnull
+    public Group getById(@PathParam("id") long id)
+    {
+        LOGGER.debug("GroupWebService.getById({})", id);
+        // Todo: auth check that caller is member of channel
+
+        return groupDao.getById(id);
+    }
+
+    @GET
+    @Path("{id}/members")
+    public List<Member> getMembers(@PathParam("id") long id)
+    {
+        Group result = groupDao.getById(id);
+        if(result != null)
+        {
+            return result.getMembers();
+        }
+
+        throw new NotFoundException();
+    }
+
+    @DELETE
+    @Path("{id}")
+    public void delete(@PathParam("id") long id)
+    {
+        LOGGER.debug("GroupWebService.delete({})", id);
+        // Todo: what is deletion logic? Any member? Only when last member quits?
+
+        Group victim = groupDao.getById(id);
+        if(victim != null) {
+            groupDao.delete(victim);
+        }
+    }
+
+    @POST
+    @Consumes(MediaType.APPLICATION_JSON)
+    public Group create(@Nonnull Group newGroup)
```

Comment:
The pattern in the channel package was to take in the mutable, possibly not completely populated `ChannelDto` entity object, and then return from the DAO layer the immutable completely populated `Channel` instance.